subgraphs/seaport/schema.graphql

```diff
@@ -0,0 +1,272 @@
+# Subgraph Schema: NFT Marketplace
+# Version: 1.0.0
+
+enum Network {
+  ARBITRUM_ONE
+  ARWEAVE_MAINNET
+  AURORA
+  AVALANCHE
+  BOBA
+  BSC # aka BNB Chain
+  CELO
+  COSMOS
+  CRONOS
+  MAINNET # Ethereum Mainnet
+  FANTOM
+  FUSE
+  HARMONY
+  JUNO
+  MOONBEAM
+  MOONRIVER
+  NEAR_MAINNET
+  OPTIMISM
+  OSMOSIS
+  MATIC # aka Polygon
+  XDAI # aka Gnosis Chain
+}
+
+enum NftStandard {
+  ERC721
+  ERC1155
+  UNKNOWN
+}
+
+enum SaleStrategy {
+  " Strategy that executes an order at a fixed price that can be taken either by a bid or an ask. "
+  STANDARD_SALE
+
+  " Strategy that executes an order at a fixed price that can be matched by any tokenId for the collection. "
+  ANY_ITEM_FROM_COLLECTION
+
+  " Strategy that executes an order at a fixed price that can be matched by any tokenId in a set of tokenIds. "
+  ANY_ITEM_FROM_SET
+
+  " Strategy to launch a Dutch Auction for a token where the price decreases linearly until a specified timestamp and end price defined by the seller. "
+  DUTCH_AUCTION
+
+  " Strategy to set up an order that can only be executed by a specific address. "
+  PRIVATE_SALE
+}
+
+type Marketplace @entity {
+  " Smart contract address of the protocol's main contract (Factory, Registry, etc) "
+  id: ID!
+
+  " Name of the NFT marketplace, for example LooksRare "
+  name: String!
+
+  " Slug of the NFT marketplace, for example looksrare "
+  slug: String!
+
+  " The blockchain network this subgraph is indexing on. "
+  network: Network!
+
+  " Version of the subgraph schema, in SemVer format (e.g. 1.0.0) "
+  schemaVersion: String!
+
+  " Version of the subgraph implementation, in SemVer format (e.g. 1.0.0) "
+  subgraphVersion: String!
+
+  " Version of the methodology used to compute metrics, loosely based on SemVer format (e.g. 1.0.0) "
+  methodologyVersion: String!
+
+  " Number of collections listed on the marketplace. "
+  collectionCount: Int!
+
+  " Trade count of the all collections on the marketplace. "
+  tradeCount: Int!
+
+  " Cumulative trade volume (in ETH) "
+  cumulativeTradeVolumeETH: BigDecimal!
+
+  " Revenue that goes to the marketplace protocol, aka protocol fee. "
+  marketplaceRevenueETH: BigDecimal!
+
+  " Revenue that goes to creator, aka royalty fee. "
+  creatorRevenueETH: BigDecimal!
+
+  " Sum of marketplaceRevenueETH and creatorRevenueETH. "
+  totalRevenueETH: BigDecimal!
+
+  " Cumulative unique traders "
+  cumulativeUniqueTraders: Int!
```

Comment:
I think we should add `collections: [Collection!]! derivedFrom marketplace` and `snapshots: [MarketplaceDailySnapshot!]! derivedFrom marketplace` here.